### ISSUE TITLE
使用triplet在相同的环境下测试无法达到正常的结果

### ISSUE THREAD

**Issue by rjquan** [KaiyangZhou/deep-person-reid#14]
您好，我按照您的环境（PyTorch (0.4.0)，torchvision (0.2.1)，Python2）以及方法复现了cross entropy loss+ triplet loss 的reid，数据集是market1501， 最后的结果只能有：
单卡  Rank-1/Rank-5/Rank-10:  81.0/92.6/95.4  mAP:63.1
多（4）卡  Rank-1/Rank-5/Rank-10:  43.2/66.7/75.8  mAP:24.2
单独复现cross entropy loss的是结果正常能达到您的结果，加上triplet loss会有以上情况出现。使用您提供的pretrained的模型测试也是能够正常达到rank-1 87的结果
请问可能是哪里有问题呢
谢谢

**Comment by KaiyangZhou** [comment]
@rjquan did you change any parameters in `train_img_model_xent_htri .py`? if you use the default settings, you would get similar results. Please try again and let me know.

**Comment by rjquan** [comment]
No, I didn't change the default settings and I tried several times. It seems like that when using triplet loss will get bad results.

**Comment by SHUHarold** [comment]
I also have tried using the above two losses on market1501 dataset, and got the same lower results (Rank1/82.9% Rank5/93.2%). It's so wonder that these results occurred, by the way, the settings used are the default.

**Comment by gceboh** [comment]
Perhaps because of the [rewriting of sampler for triplet loss](https://github.com/KaiyangZhou/deep-person-reid/releases/tag/v0.2.0) in v0.2.0 ?
Please try [v0.1.0](https://github.com/KaiyangZhou/deep-person-reid/tree/v0.1.0) ,and see whether the old version could solve this problem.

**Comment by Kewenjing1020** [comment]
@KaiyangZhou  hi, I've met the same situation. I set the hyper-parameters as default and train up to 180 epoch, but the best result I got is mAP 62.9/ rank1 80.9. Could you please give me some advice on how to reach the performance as in your benchmark?
